### ISSUE TITLE
Documentation suggestion: why is prefer-lodash-method recommended?

### ISSUE THREAD

**Issue by danvk** [wix/eslint-plugin-lodash#98]
`prefer-lodash-method` is included in the `lodash/recommended` rule set. The [docs for that rule][1] just say:

> When using native functions like forEach and map, it's often better to use the Lodash implementation.

It would be helpful if this page mentioned the reasons it's "often better" to use the lodash methods. I'm thinking about disabling this warning, but I'd like to know the pros/cons.

[1]: https://github.com/wix/eslint-plugin-lodash/blob/7c56ccae61aa54ed897d45c3419836c6591fd2bb/docs/rules/prefer-lodash-method.md

**Comment by jfmengels** [comment]
AFAIK, there are (at least) three reasons:
- Performance reasons: Lodash methods tend to be quite a bit faster than native ones.
- It "crashes" less: In the sense that `array.map(fn)` would crash when `array` is null, Lodash's methods do not (map returns empty array, forEach does nothing for instance)
- You can use shorthands that make quite a few iteratees/predicates shorter to write:
```js
const refs = array.map(function(value) {
  return value.ref;
});
const refs = array.map(value => value.ref); // Shorter arrow notation
// vs
const refs = _.map(array, 'ref'); // again, if there are `undefined` values in array, this won't crash either


const bigApple = fruits.find(function(fruit) { // Array.prototype.find exists in ES6
  return fruit.type === 'apple' && fruit.size === 'big';
});
const bigApple = fruits.find(fruit => fruit.type === 'apple' && fruit.size === 'big'); // Shorter arrow notation
// vs
const bigApple = _.find(fruits, {type: 'apple', size: 'big'});
```
